src/test/integration/emailPassword.test.ts

```diff
@@ -59,4 +61,89 @@ describe('EmailPasswordProvider', () => {
         }
         throw Error();
     });
+    describe('Reset password', () => {
+        let resetToken: string;
+        test('User can reset password', async () => {
+            const authenticator = getResetPasswordAuthenticator();
+            const { token } = await authenticator.resetPassword(email);
+            expect(token).toBeTruthy();
+            expect(token.length).toBeGreaterThan(0);
+            resetToken = token;
+        });
+        test('Non-existing user cannot reset password', async () => {
```

Comment:
I am sceptical about this feature. To behave differently if the user is present or not to avoid user scanning. You should be able to make the request, we should accept it OK, but we won't have the user so we can end the process right there.

Reply:
I am still thinking about it. Which message should we? I think the `user does not exist` message is ok. This is the only validation we are doing here :thinking: You can change this message if you define `onAuthenticationFailure` function

Reply:
I've been thinking about not throwing error at all :)

Reply:
changed